File: app/src/main/java/com/andretietz/updater/Bootstrap.java
package com.andretietz.updater;

import org.update4j.Configuration;

import java.io.FileReader;
import java.io.IOException;

public class Bootstrap {
    public static void main(String[] args) throws IOException {
        BullShit.INSTANCE.someExternalLib();
        Configuration updater = Configuration.read(new FileReader("update.xml"));
        updater.update();
        updater.launch();
    }
}

File: app/src/main/kotlin/com/andretietz/updater/Bootstrap.kt
package com.andretietz.updater

import org.update4j.Configuration
import java.io.File
import java.io.FileReader


class Bootstrap {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            println("Start Application ")
            val configuration = Configuration.read(FileReader(File("update.xml")))
            if (configuration.requiresUpdate()) {
                configuration.update()
            }
            configuration.launch()

        }
    }
}